### PR TITLE
Add ability to close the survey

### PR DESCRIPTION
**Pull request by gfting**
Closes https://app.asana.com/0/1151230758309024/1198209873692927

- Implements new School model to handle school-specific configurations

- Adds UI for school admins to close the survey
image
image

- Either renders closed fallback page or the survey
Screen Shot 2020-12-06 at 10 30 47 AM

### CODE REVIEW

File: server/src/routes/surveyRoutes.js [hunk 1]
```diff
@@ -112,6 +114,102 @@ router.get('/allResponses', passport.authenticate('jwt', { session: false }), as
...
+ */
+router.post('/closeDate/:school', passport.authenticate('jwt', { session: false }), async (req, res) => {
+    const { school } = req.params;
+    const { role, school: userSchool } = req.user;
+    const { closeDate } = req.body;
+
+    if (role === roles.schoolAdmin && userSchool === school && schoolsArray.includes(userSchool)) {
+        try {
+            // // Get the school's appropriate school model
+            // const castedDate = new Date(closeDate).toISOString(); // convert to date
```
Comment (Lynx1820) [inline server/src/routes/surveyRoutes.js:184]: Is this meant to be commented?

File: client/src/utils/index.js [hunk 2]
```diff
@@ -199,4 +199,45 @@ export const unsubscribe = (token) => (
...
-)
\ No newline at end of file
+)
+
+/**
+ * Handles setting the sender email for a school
+ * 
+ * @param {string} school – user school 
+ */
+export const changeCloseDate = (data, school) => (
```
Comment (Lynx1820) [inline client/src/utils/index.js:209]: What happens when you try to change the closing date after the survey has already been closed? From testing, it seems like it doesn't reopen properly.

**Review by Lynx1820 — APPROVED** [review]
LGTM, just the one comment. I think it's non-blocking and can be resolved in another PR.

File: server/src/routes/surveyRoutes.js (hunk 1 above)
Comment (gfting) [reply server/src/routes/surveyRoutes.js:184]: Yes, it's an alternate pattern, as the Mongoose method might be deprecated soon

File: client/src/utils/index.js (hunk 2 above)
Comment (gfting) [reply client/src/utils/index.js:209]: This was resolved in another ticket, it's an issue with the token and survey link being generated, not with this PR.

**Pull request merged**